python_modules/dagster/dagster/config/field_utils.py

```diff
@@ -164,30 +166,35 @@ def partially_specified_config(context) -> List:
             return sorted(list(context.op_config.items()))
     """
 
-    def __init__(self, key_type, inner_type):
+    def __init__(self, key_type, inner_type, name=None):
         from .field import resolve_to_config_type
 
         self.key_type = resolve_to_config_type(key_type)
         self.inner_type = resolve_to_config_type(inner_type)
+        self.given_name = name
 
         check.inst_param(self.key_type, "key_type", ConfigType)
         check.inst_param(self.inner_type, "inner_type", ConfigType)
         check.param_invariant(
             self.key_type.kind == ConfigTypeKind.SCALAR, "key_type", "Key type must be a scalar"
         )
+        check.opt_str_param(self.given_name, "name")
 
         super(Map, self).__init__(
-            key="Map.{key_type}.{inner_type}".format(
-                key_type=self.key_type.key, inner_type=self.inner_type.key
+            key="Map.{key_type}.{inner_type}{name_key}".format(
+                key_type=self.key_type.key,
+                inner_type=self.inner_type.key,
+                name_key=f":name: {name}" if name else "",
             ),
+            given_name=name,
```

Comment:
this is a bit clever using this field on the parent for giving custom types names for storing the key label. 

definitely deserves some comments